frontend/src/pages/surveys/[id].tsx

```diff
@@ -0,0 +1,95 @@
+import { Container, Content, Layout } from 'styles/main.styles';
+import { Box, Button, Heading, Spinner, Tooltip } from '@chakra-ui/react';
+import Head from 'next/head';
+import { Sidemenu } from '@components/sidemenu';
+import { useRouter } from 'next/router';
+import { SurveyService } from '@api/services/survey';
+import { httpClient } from '@api/client';
+import { useEffect, useState } from 'react';
+import { Survey } from 'api/models/survey';
+import { GetServerSideProps } from 'next';
+import { AUTH_COOKIE } from 'consts';
+import { parseCookies } from 'nookies';
+import { HiArrowLeft } from 'react-icons/hi';
+import { SurveyHandler } from '@components/survey-handler';
+
+const surveyService = new SurveyService(httpClient);
+
+const SurveyPage = () => {
+  const router = useRouter();
+  const { id } = router.query;
+  const [survey, setSurvey] = useState<Survey>();
+  const [loading, setLoading] = useState(false);
+
+  const loadSurvey = async () => {
+    setLoading(true);
+    const response = await surveyService.get(Number(id));
+    setSurvey(response.data);
+    setLoading(false);
+  };
```

Comment:
I suggest wrapping this block in a try/catch/finally, and call the `setLoading(false)` inside the `finally`